### PR TITLE
CB-4748 In case of environments with CCM, the 9443 port should not be opened for control plane

### PR DESCRIPTION
**Pull request by lacikaaa**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/GatewaySecurityGroupWithDefaultGatewayCidrsDecorator.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import com.sequenceiq.cloudbreak.domain.SecurityGroup;
+import com.sequenceiq.cloudbreak.domain.SecurityRule;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceGroup;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+import com.sequenceiq.common.api.type.Tunnel;
+
+@Component
+public class GatewaySecurityGroupWithDefaultGatewayCidrsDecorator {
```
Comment (keyki) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/GatewaySecurityGroupWithDefaultGatewayCidrsDecorator.java]: Can we use a simpler name? Like GatewaySecurityGroupDecorator?

Comment (lacikaaa) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/stack/GatewaySecurityGroupWithDefaultGatewayCidrsDecorator.java]: renamed

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/GatewaySecurityGroupDecorator.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Value("${cb.https.port}")
+    private Integer httpsPort;
+
+    public void extendGatewaySecurityGroupWithDefaultGatewayCidrs(Stack stack, Tunnel tunnel) {
+        Set<InstanceGroup> gateways = filterForGateways(stack);
+        Set<String> defaultGatewayCidrs = filterEmptyCidrs();
+        LOGGER.debug("Extending security group for gateways: [{}] DefaultGatewayCidrs: [{}] Tunnel: [{}]",
+                gateways.stream().map(InstanceGroup::getGroupName).collect(Collectors.joining(", ")), defaultGatewayCidrs, tunnel);
+        if (!defaultGatewayCidrs.isEmpty() && !tunnel.useCcm()) {
```
Comment (keyki) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/GatewaySecurityGroupDecorator.java]: I think we should use stack.getUseCcm it handles if the tunnel is not set.

File: freeipa/src/main/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverter.java [hunk 3]
```diff
@@ -121,7 +121,7 @@ private void extendGatewaySecurityGroupWithDefaultGatewayCidrs(Stack stack) {
         Set<InstanceGroup> gateways =
                 stack.getInstanceGroups().stream().filter(ig -> InstanceGroupType.MASTER == ig.getInstanceGroupType()).collect(Collectors.toSet());
         Set<String> defaultGatewayCidrs = defaultGatewayCidr.stream().filter(StringUtils::isNotBlank).collect(Collectors.toSet());
-        if (!defaultGatewayCidrs.isEmpty()) {
+        if (!defaultGatewayCidrs.isEmpty() && !stack.getTunnel().useCcm()) {
```
Comment (keyki) [inline freeipa/src/main/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverter.java]: Same stack.getUseCcm()

**Pull request merged**